### PR TITLE
#74 count in filter throws not supported

### PR DESCRIPTION
**Pull request by Leandro-Albano**
- Added `QueryNodeKind.Count` as a valid option on  `ValidateSingleValueNode`
- Added e2e tests for $count in $filter
- Fixed missing Id's in customer list on ODataRoutingSample

### CODE REVIEW

File: test/Microsoft.AspNetCore.OData.E2E.Tests/Query/ActionResult/ActionResultTests.cs [hunk 1]
```diff
@@ -89,5 +89,30 @@ public async Task ActionResultODataPathReturnsBaseWithoutExpansion()
...
+        /// <summary>
+        /// $filter should work with $count.
+        /// </summary>
+        [Theory]
+        [InlineData(1, 1)]
+        [InlineData(0, 0)]
+        public async Task ActionResultShouldAllowCountInFilter(int filterParam, int expectedItemsCount)
+        {
+            // Arrange
+            string queryUrl = string.Format($"odata/Customers?$filter=Books/$count eq {filterParam}");
```
Comment (xuzhg) [inline test/Microsoft.AspNetCore.OData.E2E.Tests/Query/ActionResult/ActionResultTests.cs:102]: >string.Format [](start = 30, length = 13)

It seems we don't need string.Format here

**Comment by xuzhg** [comment]
>             string queryUrl = string.Format("odata/Customers");

would you please help to clear others in this file?

---
Refers to: test/Microsoft.AspNetCore.OData.E2E.Tests/Query/ActionResult/ActionResultTests.cs:76 in 1a4f57b. [](commit_id = 1a4f57b732b0091167fcacc1750b24196758911c, deletion_comment = False)

**Review by xuzhg — APPROVED** [review]
:shipit:

**Comment by Leandro-Albano** [comment]
Good point! I added another commit.

**Pull request merged**